charts/scalardb-mcp-server/templates/scalardb-mcp-server/ingress.yaml

```diff
@@ -0,0 +1,62 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "scalardb-mcp-server.fullname" . -}}
+{{- $svcPort := .Values.scalardbMcpServer.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
```

Comment:
Sorry if I miss something, but I'd like to know what the purpose of this logic is. Could you please elaborate on the use case of this if statement?